pkgs/by-name/cl/clevis/package.nix

```
@@ -0,0 +1,125 @@
+{
+  lib,
+  asciidoc,
+  bash,
+  coreutils,
+  cryptsetup,
+  curl,
+  fetchFromGitHub,
+  gnugrep,
+  gnused,
+  jansson,
+  jose,
+  jq,
+  libpwquality,
+  luksmeta,
+  makeWrapper,
+  meson,
+  ninja,
+  nixosTests,
+  pkg-config,
+  stdenv,
+  tpm2-tools,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "clevis";
+  version = "20";
+
+  src = fetchFromGitHub {
+    owner = "latchset";
+    repo = "clevis";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-rBdZrnHPzRd9vbyl1h/Nb0cFAtIPUHSmxVoKrKuCrQ8=";
+  };
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  patches = [
+    # Replaces the clevis-decrypt 300s timeout to a 10s timeout
+    # https://github.com/latchset/clevis/issues/289
+    ./0001-tang-timeout.patch
+  ];
+
+  postPatch = ''
+    for f in $(find src/ -type f); do
+      grep -q "/bin/cat" "$f" && \
+      substituteInPlace "$f" \
+        --replace-fail '/bin/cat' '${coreutils}/bin/cat' || true
+    done
```

What's the point of using `--replace-fail` if eventually you use `|| true`? I noticed that you just moved the expression, and wanted to tide it up etc, so I'm sorry that someone else's bad quality Bash writing responsibility falls upon you, but I must express an alternative approach to this:

```suggestion
    find src/ -type f -exec grep -q "/bin/cat" {} \; | while read f; do
      substituteInPlace "$f" \
        --replace-fail '/bin/cat' '${coreutils}/bin/cat'
    done
```